web/src/search/input/interactive/FilterInput.tsx

```diff
@@ -0,0 +1,416 @@
+import { ComponentSuggestions, noSuggestions, typingDebounceTime } from '../QueryInput'
+import { isDefined } from '../../../../../shared/src/util/types'
+import Downshift from 'downshift'
+import { generateFiltersQuery } from '../helpers'
+import { QueryState, interactiveFormatQueryForFuzzySearch } from '../../helpers'
+import { dedupeWhitespace } from '../../../../../shared/src/util/strings'
+import { FiltersToTypeAndValue } from '../../../../../shared/src/search/interactive/util'
+import { SuggestionTypes } from '../../../../../shared/src/search/suggestions/util'
+import { startCase } from 'lodash'
+import { searchFilterSuggestions } from '../../searchFilterSuggestions'
+import { LoadingSpinner } from '@sourcegraph/react-loading-spinner'
+import { CheckButton } from './CheckButton'
+
+interface Props {
+    /**
+     * The filters currently added to the query.
+     */
+    filtersInQuery: FiltersToTypeAndValue
+
+    /**
+     * The query in the main query input.
+     */
+    navbarQuery: QueryState
+
+    /**
+     * The key representing this filter in the top-level filtersInQuery map.
+     */
+    mapKey: string
+
+    /**
+     * The value for this filter.
+     */
+    value: string
+
+    /**
+     * The search filter type, as available in {@link SuggstionTypes}
+     */
+    filterType: SuggestionTypes
+
+    /**
+     * Whether or not this FilterInput is currently editable.
+     *
+     * This is passed as a prop rather than being a state field because
+     * this component is unaware whether to render as editable or uneditable
+     * on initial mount.
+     */
+    editable: boolean
+
+    /**
+     * Callback that handles a filter input being submitted. Triggers a search
+     * with the new query value.
+     */
+    onSubmit: (e: React.FormEvent<HTMLFormElement>) => void
+
+    /**
+     * Callback to handle the filter's value being updated.
+     */
+    onFilterEdited: (filterKey: string, value: string) => void
+
+    /**
+     * Callback to handle the filter chip being deleted.
+     */
+    onFilterDeleted: (filterKey: string) => void
+
+    /**
+     * Callback to handle the editable state of this filter.
+     */
+    toggleFilterEditable: (filterKey: string) => void
+}
+
+const LOADING: 'loading' = 'loading'
+
+interface State {
+    /** Only show suggestions if search input is focused */
+    showSuggestions: boolean
+    /** The suggestions shown to the user */
+    suggestions: typeof LOADING | ComponentSuggestions
+
+    /** The value being typed in the filter's input*/
+    inputValue: string
+}
+
+/**
+ * The filter chips for each filter added to the query in interactive mode. The filter input can be either editable or non-editable.
+ * If it's in an editable state, it consists of a text input field, with suggestions, and a close button. Otherwise, it's a simple
+ * button with a close button.
+ */
+export default class FilterInput extends React.Component<Props, State> {
```

Comment:
```suggestion
export class FilterInput extends React.Component<Props, State> {
```
We generally avoid default exports for components